### PR TITLE
Milestone 2: Add project structure

### PR DESCRIPTION
**Pull request by descholar-ceo**
## What does this PR do?
It create the basic project structure

## Tasks to be completed on this PR
- [x] Prepare the directory structure
- [x] Implement the App component (with redux)
- [x] Implement the BooksList component
- [x] Implement the Book component
- [x] Implement the BooksForm component
- [x] Define the Redux actions and reducers
- [x] Initialize the Redux store

## Deployment
[See this BookStore on heroku](https://descholar-bookstore-react.herokuapp.com/)

### CODE REVIEW

File: src/components/Book.js
Comment (christianotieno) [inline src/components/Book.js]: ## Linter Disabled ⚠️ 
- [x] There is an easy fix for this rule instead of the disable option. Let's lift this and improve on our propTypes definition, thank you!

File: src/containers/BookList.js
Comment (christianotieno) [inline src/containers/BookList.js]: ## Linter Disabled ⚠️ 
- [x] There is an easy fix for this rule instead of the disable option. Let's lift this and improve on our propTypes definition, thank you!

File: src/containers/BooksForm.js
Comment (christianotieno) [inline src/containers/BooksForm.js:2]: ## Linter Disabled ⚠️ 
- [ ] There is an easy fix for this rule instead of the disable option. Let's lift this and improve on our unused variables usage, thank you!

**Review by christianotieno — APPROVED** [review]
### Milestone Approved 🍾 
Hello @descholar-ceo 👋 
 ##
This milestone has met nearly all the requirements needed for approval. 👏 
Congratulations on the hard work done on this milestone. Good job 👍 
I will go ahead and mark this as complete, but before merging, take note on the suggested optional changes below 👇 

Good luck with the next tasks 🥇 
##
### Happy coding 💻 
#### Christian Otieno (TSE)

File: src/containers/BookList.js
Comment (descholar-ceo) [reply src/containers/BookList.js]: Fixed this

File: src/components/Book.js
Comment (descholar-ceo) [reply src/components/Book.js]: Fixed this as well

File: src/containers/BooksForm.js
Comment (descholar-ceo) [reply src/containers/BooksForm.js:2]: I think I will fix this in the next milestone, when I will start to use that array.

**Pull request merged**
